AsyncDisplayKit/ASMapNode.mm

```diff
@@ -180,7 +171,9 @@ - (void)takeSnapshot
 - (void)setUpSnapshotter
 {
   ASDisplayNodeAssert(!CGSizeEqualToSize(CGSizeZero, self.calculatedSize), @"self.calculatedSize can not be zero. Make sure that you are setting a preferredFrameSize or wrapping ASMapNode in a ASRatioLayoutSpec or similar.");
-  _options.size = self.calculatedSize;
+  if (!_options) {
+    [self createInitialOptions];
+  }
   _snapshotter = [[MKMapSnapshotter alloc] initWithOptions:_options];
```

Comment:
Normally I prefer accessing instance variables and so as you have this written is probably more efficient, but because this won't be called frequently at all, maybe we should do self.options here and delete the nil check.  Then in the options accessor method, we could make sure that a nil options has the default options created for it.